src/screens/NFTList.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag'
+import BackButton from '../components/BackButton';
+import { NFT_LIST_PER_PAGE } from '../constants';
+import HathorPaginate from '../components/HathorPaginate';
+import NFTListElement from '../components/NFTListElement';
+import helpers from '../utils/helpers';
+import { connect } from "react-redux";
+
+
+const mapStateToProps = (state) => {
+  const registeredTokens = state.tokens;
+  const tokenMetadata = state.tokenMetadata || {};
+  const nftList = Object.values(tokenMetadata).filter((meta) => {
+    const hasNFTMedia = meta.nft_media && meta.nft_media.file;
+    const isRegistered = state.tokens.find((registeredToken) => registeredToken.uid === meta.id);
```

Comment:
This is an O(n^2) operation. Should we index in O(n) and check in O(1), reducing the overall complexity to O(n)? I feel the code change is simple enough to do this refactor.

Reply:
Actually this was not needed anymore, given that we are now only storing metadata of registered tokens. Removed this part here: https://github.com/HathorNetwork/hathor-wallet/pull/228/commits/96bb6e497a99d75557e1b89587c624677e15dbd5

Reply:
Ok. Thanks!